profiles/views.py

```diff
@@ -55,3 +56,36 @@ def get_serializer_class(self):
             return self.serializer_class_limited
         # Case #5: this should never happen, but just in case
         return self.serializer_class_private
+
+    def get_object(self):
```

Comment:
Ok, but I think this job needs to be done in the Permission classes and not here.